profiler/profiler_test.go

```diff
@@ -126,6 +126,20 @@ func TestStart(t *testing.T) {
 		mu.Lock()
 		mu.Unlock()
 	})
+
+	t.Run("aws-lambda", func(t *testing.T) {
+		orig := functionName
+		functionName = "my-function-name"
+		defer func() { functionName = orig }()
```

Comment:
If we switch to checking the env var when starting the profiler, we can do this instead:

`t.Setenv("AWS_LAMBDA_FUNCTION_NAME", "my-function-name")`

This will set the env var for the duration of the test, and restore it/unset it as needed when the test case is done.